src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -277,35 +394,117 @@ public override void ClearAssetCache()
+                    oldAssets[assetId.Name].Add(assetId);
+                }
+
+                var entry = new AssetCacheEntry();
+                entry.Asset = asset;
+                entry.AssetId = assetId;
+                lookupByAsset[asset] = entry;
+                lookupByAssetId[assetId] = entry;
             }
 
-            CleanUpUnused(oldAssets, unvisitedAssets);
-            assets = oldAssets.Values.OrderBy(x => x.AssetId.FileIdentifier).ThenBy(x => x.AssetId.Guid).ToArray();
+            foreach(var idPair in unvisitedIds)
+            {
+                if (oldAssets.TryGetValue(idPair.Item1, out List<AssetId> ids) &&
+                    ids.Contains(idPair.Item2))
+                {
+                    ids.Remove(idPair.Item2);
+                }
+            }
 
+            lookupByName = oldAssets;
+            var tempAssets = new List<NameEntry>();
+            foreach(var item in lookupByName)
+            {
+                tempAssets.Add(new NameEntry(item.Key, item.Value.OrderBy(a => a.Guid).ThenBy(a => a.FileIdentifier).ToArray()));
+            }
+            assets = tempAssets.OrderBy(a => a.Name).ToArray();
             EditorUtility.SetDirty(this);
 #endif
         }
+
+        public virtual void SaveAssets()
+        {
+#if UNITY_EDITOR
+            EnsureAssetDirectoryExists();
+            if (assets == null ||
+                assets.Length == 0)
+            {
+                Debug.Log("Assets was null or empty, rerunning asset cache update");
+                UpdateAssetCache();
+            }
+
+            if (assets == null ||
+                assets.Length == 0)
+            {
+                Debug.Log("No assets were found.");
+                return;
+            }
+
+            foreach(var nameEntry in assets)
+            {
+                string assetName = GetAssetCachesContentPath($"{GetValidAssetName(nameEntry.Name)}_{this.GetType().Name}", assetContentExtension);
+                AssetCacheContent content = ScriptableObject.CreateInstance<AssetCacheContent>();
+                List<AssetCacheEntry> assetCacheEntries = new List<AssetCacheEntry>();
+                foreach (var assetId in nameEntry.Ids)
+                {
+                    if (!lookupByAssetId.TryGetValue(assetId, out var entry))
+                    {
+                        Debug.LogError($"AssetId did not have a registered Asset: {assetId}");
+                        continue;
+                    }
+
+                    assetCacheEntries.Add(entry);
+                }
+
+                content.AssetCacheEntries = assetCacheEntries.ToArray();
+                Debug.Log($"Creating asset: {content} {assetName}");
+                AssetDatabase.CreateAsset(content, assetName);
+            }
+#endif
+        }
+
+        string GetValidAssetName(string name)
+        {
+            if (name == null)
+            {
+                return null;
+            }
+
+            return name.Replace(":", "_");

```

Comment:
>return name.Replace(":", "_"); [](start = 12, length = 30)

This needs to replace any invalid path character in a name with some other valid character, including "/", "\", etc. You can use Path.GetInvalidPathChars to get the complete list